### PR TITLE
Setup DELETE API for Hard Deletion of User

### LINKED ISSUE #10: [BE] [User Service] Implement DELETE API for user removal

### PR DESCRIPTION
**Pull request by shishirbychapur**
Closes #10 

~~Note: unsure whether we are soft deleting or hard deleting, but since there is a `deletedAt` field in the user dto, i have assumed that it is the former. Please correct me if i'm wrong.~~

Also added `CODEOWNERS` file to automate the process of selecting reviewers for the PRs.

### CODE REVIEW

**Comment by glemenneo** [comment]
I think for now we can do hard delete so that we dont have to handle edge cases for get, post and put apis

**Comment by Daviancold** [comment]
Not sure if it is necessary, but would we need to handle the case where we try deleting a user that does not exist?

**Comment by glemenneo** [comment]
Regarding the case of deleting a user that does not exist, I've opened a new pr that automatically deals with such errors.

File: backend/user-service/src/models/user.repository.ts [hunk 1]
```diff
@@ -32,6 +32,6 @@ export async function updateUser(id: string, dto: UserDto | UserProfileDto): Pro
     return userModel.findByIdAndUpdate(id, dto, { new: true })
 }
 
-export async function deleteUser(id: string): Promise<void> {
-    await userModel.findByIdAndDelete(id)
+export async function deleteUser(id: string): Promise<IUser | null> {
+    return userModel.findByIdAndDelete(id)
```
Comment (glemenneo) [inline backend/user-service/src/models/user.repository.ts:36]: do we need to return the deleted user?

Comment (shishirbychapur) [reply backend/user-service/src/models/user.repository.ts:36]: not really i guess, it was more for testing purposes and verification

**Review by glemenneo — APPROVED** [review]
lgtm

**Review by Daviancold — APPROVED** [review]

**Pull request merged**